src/CompanyService.Validation/CompanyUser/CreateCompanyUserRequestValidator.cs

```diff
@@ -12,33 +12,40 @@ public class CreateCompanyUserRequestValidator : AbstractValidator<CreateCompany
   {
     public CreateCompanyUserRequestValidator(
       ICompanyRepository companyRepository,
+      ICompanyUserRepository companyUserRepository,
       IContractSubjectRepository contractRepository,
       IUserService userService)
     {
-      When(x => x.Rate.HasValue, () =>
-      {
-        RuleFor(x => x.Rate)
-        .Must(x => x.Value > 0 && x.Value <= 1)
-        .WithMessage("Rame must be from 0 to 1.");
-      });
+      RuleFor(x => x.UserId)
+        .MustAsync(async (userId, _) => !await companyUserRepository.DoesExistAsync(userId))
+        .WithMessage("User already exists.")
+        .DependentRules(() =>
+        {
+          When(x => x.Rate.HasValue, () =>
+          {
+            RuleFor(x => x.Rate)
+              .Must(x => x.Value > 0 && x.Value <= 1)
+              .WithMessage("Rate must be from 0 to 1.");
+          });
 
-      RuleFor(x => x.ContractTermType)
-        .IsInEnum();
+          RuleFor(x => x.ContractTermType)
```

Comment:
fix tab

Reply:
it is in DependentRules block, so it's ok i think